saleor/graphql/order/tests/test_order.py

```diff
@@ -6784,6 +6785,133 @@ def test_draft_order_query_with_filter_created_(
     assert len(orders) == count
 
 
+@pytest.mark.parametrize("is_click_and_collect", [True, False])
+def test_order_query_with_filter_is_click_and_collect(
+    is_click_and_collect,
+    orders_query_with_filter,
+    staff_api_client,
+    permission_manage_orders,
+    orders,
+    other_channel_USD,
+    warehouse_for_cc,
+):
+    # given
+    order_1 = orders[0]
+    order_1.collection_point = warehouse_for_cc
+    order_1.collection_point_name = warehouse_for_cc.name
+
+    order_2 = orders[1]
+    order_2.collection_point_name = warehouse_for_cc.name
+
+    order_3 = orders[2]
+    order_3.collection_point = warehouse_for_cc
+
+    cc_orders = [order_1, order_2, order_3]
+
+    Order.objects.bulk_update(cc_orders, ["collection_point", "collection_point_name"])
+
+    variables = {"filter": {"isClickAndCollect": is_click_and_collect}}
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    returned_orders = content["data"]["orders"]["edges"]
+    if is_click_and_collect:
```

Comment:
Don't we want to create two separate tests?

Reply:
I'm with @L3str4nge here. Conditional logic inside the test body is the best way to tell multiple test cases are pretending to be one.

Reply:
Yeah but `given` and `when` could be easily wrapped as `fixture` and reuse. Actually this test does not test one thing but two separate scenarios.